test/select.jl

```diff
@@ -899,24 +899,24 @@ end
     @test df2 ≅ DataFrame(x1_first=1, x2_first=1, x3_first=missing,
                           x4_first=missing)
     @test df2.x1_first isa Vector{Int}
-    @test df2.x2_first isa CategoricalVector
+    @test df2.x2_first isa CategoricalVector{Int}
     @test df2.x3_first isa Vector{Missing}
     @test df2.x4_first isa Vector{Missing}
```

Comment:
This result isn't ideal, as levels are lost. But I guess there's no general solution to this.

Reply:
I think the decision how this should be handled should be in CategoricalArrays.jl not here - here we should use the mechanics exposed there. Do you agree?

Reply:
I'm not sure it can be fixed, even there. Basically it's due to the fact that we allocate the column based on the type of the returned value, not based on the possible types that could have been returned. That's the same problem as with `[v for v in x]` when `x` happens to contain only missing values. We'll have to live with it, or move to `DataValue{Union{Missing, CategoricalValue{Int}}`.

Reply:
Ah - right